examples/platforms/efr32/uart.c

```diff
@@ -153,35 +145,32 @@ otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength)
 void processReceive(void)
 {
     // Copy tail to prevent multiple reads
-    uint16_t tail = sReceive.mTail;
+    uint16_t tail = sReceiveFifo.mTail;
 
     // If the data wraps around, process the first part
-    if (sReceive.mHead > tail)
+    if (sReceiveFifo.mHead > tail)
     {
-        otPlatUartReceived(sReceive.mBuffer + sReceive.mHead, kReceiveBufferSize - sReceive.mHead);
+        otPlatUartReceived(sReceiveFifo.mBuffer + sReceiveFifo.mHead, kReceiveFifoSize - sReceiveFifo.mHead);
 
         // Reset the buffer mHead back to zero.
-        sReceive.mHead = 0;
+        sReceiveFifo.mHead = 0;
     }
 
     // For any data remaining, process it
-    if (sReceive.mHead != tail)
+    if (sReceiveFifo.mHead != tail)
     {
-        otPlatUartReceived(sReceive.mBuffer + sReceive.mHead, tail - sReceive.mHead);
+        otPlatUartReceived(sReceiveFifo.mBuffer + sReceiveFifo.mHead, tail - sReceiveFifo.mHead);
 
         // Set mHead to the local tail we have cached
-        sReceive.mHead = tail;
+        sReceiveFifo.mHead = tail;
     }
 }
 
 void processTransmit(void)
 {
     otEXPECT(sTransmitBuffer != NULL);
 
-    for (; sTransmitLength > 0; sTransmitLength--)
-    {
-        USART_Tx(USART0, *sTransmitBuffer++);
-    }
+    UARTDRV_TransmitB(sUartHandle, (uint8_t *)sTransmitBuffer, sTransmitLength);
```

Comment:
Why not use non blocking UARTDRV_Transmit? As I see OT is not designed to be blocked in process drivers call. I think sending big buffer may introduce RF packet drops etc. I notice some issue with EFR32 in case intensive data exchange and sending logs via uart. It looks like the root cause was delay in processTransmit but I didn't investigate it so much.

Reply:
I'll switch this to non-blocking soon.

Reply:
Updated to use non-blocking `UARTDRV_Transmit`.